src/Sparrow/Json/JsonContextPoolBase.cs

```diff
@@ -91,21 +120,58 @@ private void EnsureCurrentThreadContextWillBeReleased(int currentThreadId)
                     {
                         if(Interlocked.CompareExchange(ref copy[i].ThreadId, currentThreadId, -1) == -1)
```

Comment:
I think that the problem with `CleanThreadLocalState` still exists? https://github.com/ppekrol/ravendb/commit/99acb4e904c3938bd1845300b3e67c96d1da98f0#diff-1aa997bd6d5750decc752ea2c476347bR186

Reply:
@ppekrol I've run the slowTests-> kill in the middle scenario, and it worked well

Reply:
yes, we should call the ContextStackThreadReleaser release logic from there, assign null and suppress finalization